### ISSUE TITLE
Sticky toolbar

### ISSUE THREAD

**Issue by fredck** [ckeditor/ckeditor5-build-decoupled-document#7]
Don't know where to report this precisely.

The current [Document Editor demo](https://ckeditor.com/ckeditor-5-builds/#document) should have a sticky toolbar, just like the Classic Editor. Without that, the editor is hardly usable.

**Comment by Reinmar** [comment]
cc @oleq?

**Comment by oleq** [comment]
I was wondering the same when implementing it but I decided this is a different use–case. 

A document editor should be similar to the native word processors: It's the document (page) area that should scroll, but never the toolbar (or page). That's why the toolbar should remain static and the demo we have ATM on our website is... wrong. 

OTOH... I see nothing wrong with using the sticky toolbar in this situation. I mean: it won't break anything. It's just about the complexity of the code.

**Comment by Reinmar** [comment]
> It's just about the complexity of the code.

What if we use `position:sticky`?

**Comment by oleq** [comment]
No IE11, that's for sure. But this is [not an issue](https://caniuse.com/#feat=css-sticky). I'm not also sure what this boundary situation would look like with the native `pos: sticky`:

![image](https://user-images.githubusercontent.com/1099479/38487448-6e9d3310-3be0-11e8-8342-f67ea29c5807.png)

We make sure the toolbar won't cover the last 1-2 LOC. Is the native `pos: sticky` capable of such thing?

**Comment by fredck** [comment]
> A document editor should be similar to the native word processors: It's the document (page) area that should scroll, but never the toolbar (or page). That's why the toolbar should remain static and the demo we have ATM on our website is... wrong.

I agree with you, @oleq. I have hacked the demo a bit and got the following, which I believe should be the right demo:

![image](https://user-images.githubusercontent.com/630060/38487513-b37d0988-3be0-11e8-9ca1-ce68de72e46d.png)

**Comment by oleq** [comment]
> We make sure the toolbar won't cover the last 1-2 LOC. Is the native pos: sticky capable of such thing?

Another thing: I was unable to use the `viewportTopOffset` with `pos: sticky`. Setting `top: 123px` always moves the element, not only when it's sticky.

**Comment by Reinmar** [comment]
> Another thing: I was unable to use the viewportTopOffset with pos: sticky. Setting top: 123px always moves the element, not only when it's sticky.

I don't understand how `viewportTopOffset` is related to `pos:sticky`.  The former is our JS implementation and is not available (?) in the decoupled editor. The latter is CSS.

**Comment by oleq** [comment]
Our sticky toolbar implementation offers `viewportTopOffset`. We use it everywhere (demos, docs...).

My point was that if we decide to switch to the native `position: sticky`, we're losing this feature.

**Comment by fredck** [comment]
Being this a demo issue, I believe @m-kr is the one to call for action ;)

**Comment by Reinmar** [comment]
> My point was that if we decide to switch to the native position: sticky, we're losing this feature.

Do we? What about `position: sticky; top: 33px`? Doesn't it have the same behaviour?

**Comment by oleq** [comment]
> Do we? What about position: sticky; top: 33px? Doesn't it have the same behaviour?

[No.](https://jsfiddle.net/oleq/umy5pnmq/3/)

**Comment by m-kr** [comment]
This solution with the scrollable editable area is the best, I can implement it without any hacks and, as @oleq wrote, it looks like Word 👍
